registry/remote/repository.go

```diff
@@ -614,6 +615,9 @@ type blobStore struct {
 
 // Fetch fetches the content identified by the descriptor.
 func (s *blobStore) Fetch(ctx context.Context, target ocispec.Descriptor) (rc io.ReadCloser, err error) {
+	if isForeignLayer(target) {
+		return nil, fmt.Errorf("%s: %w", target.Digest, errdef.ErrForeignLayer)
+	}
```

Comment:
Should this check be here, or inside its upstream caller (copy)? It affects what error message to display.

Reply:
Put it here is fine.

Reply:
It's the right place. We should also apply the same logic to `blobStore.FetchReference()`.

Reply:
Besides, can we have some small unit tests?

Reply:
Moved it down after getting 404.

Reply:
After discussion, removed the check inside `Fetch()`.